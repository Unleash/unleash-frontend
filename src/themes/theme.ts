import { createTheme } from '@mui/material/styles';
import { colors } from './colors';

export default createTheme({
    boxShadows: {
        main: '0px 2px 4px rgba(129, 122, 254, 0.2)',
    },
    typography: {
        fontFamily: 'Sen, Roboto, sans-serif',
        fontWeightBold: '700',
        fontWeightMedium: '700',
        allVariants: { lineHeight: 1.4 },
        button: { lineHeight: 1.75 },
        h1: {
            fontSize: '1.5rem',
            lineHeight: 1.875,
        },
    },
    fontSizes: {
        mainHeader: '1.25rem',
        subHeader: '1.1rem',
        bodySize: '1rem',
        smallBody: '0.9rem',
        smallerBody: '0.8rem',
    },
    fontWeight: {
        thin: 300,
        medium: 400,
        semi: 700,
        bold: 700,
    },
    shape: {
        borderRadius: '3px',
        borderRadiusLarge: '10px',
        borderRadiusExtraLarge: '25px',
    },
    palette: {
        primary: {
            main: colors.purple[800],
            light: colors.purple[700],
            dark: colors.purple[900],
        },
        secondary: {
            main: colors.purple[800],
            light: colors.purple[700],
            dark: colors.purple[900],
        },
        info: {
            light: colors.blue[700],
            main: colors.blue[700],
            dark: colors.blue[800],
        },
        success: {
            light: colors.green[700],
            main: colors.green[700],
            dark: colors.green[800],
        },
        warning: {
            light: colors.orange[700],
            main: colors.orange[700],
            dark: colors.orange[800],
        },
        error: {
            light: colors.red[700],
            main: colors.red[700],
            dark: colors.red[800],
        },
        divider: colors.grey[300],
        dividerAlternative: colors.grey[500],
        grey: colors.grey,
        text: {
            primary: colors.grey[900],
            secondary: colors.grey[800],
            disabled: colors.grey[600],
        },
        code: {
            main: '#0b8c8f',
            diffAdd: 'green',
            diffSub: 'red',
            diffNeutral: 'black',
            edited: 'blue',
            background: '#efefef',
        },
        activityIndicators: {
            unknown: colors.grey[100],
            recent: colors.green[100],
            inactive: colors.orange[200],
            abandoned: colors.red[200],
        },
    },
    components: {
        MuiTableHead: {
            styleOverrides: {
                root: {
                    background: colors.grey[200],
                },
            },
        },
        MuiLink: {
            styleOverrides: {
                root: {
                    color: colors.purple[900],
                },
            },
        },
        MuiBreadcrumbs: {
            styleOverrides: {
                root: {
                    '& a': {
                        color: colors.purple[900],
                    },
                },
            },
        },
        MuiTableRow: {
            styleOverrides: {
                root: {
                    '&.MuiTableRow-hover:hover': {
                        background: colors.grey[100],
                    },
                },
            },
        },
    },
});
