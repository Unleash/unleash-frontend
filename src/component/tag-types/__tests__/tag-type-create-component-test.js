import React from 'react';
import { ThemeProvider } from '@material-ui/core';
import TagTypes from '../form-tag-type-component';
import renderer from 'react-test-renderer';
import theme from '../../../themes/main-theme';

test('renders correctly for creating', () => {
    const tree = renderer
        .create(
            <>
                <ThemeProvider theme={theme}>
                    <TagTypes
                        history={{}}
                        title="Add tag type"
                        createTagType={jest.fn()}
                        validateName={() => Promise.resolve(true)}
                        hasPermission={() => true}
                        tagType={{ name: '', description: '', icon: '' }}
                        editMode={false}
                        submit={jest.fn()}
                    />
                </ThemeProvider>
            </>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('it supports editMode', () => {
    const tree = renderer
        .create(
            <>
                <ThemeProvider theme={theme}>
                    <TagTypes
                        history={{}}
                        title="Add tag type"
                        createTagType={jest.fn()}
                        validateName={() => Promise.resolve(true)}
                        hasPermission={() => true}
                        tagType={{ name: '', description: '', icon: '' }}
                        editMode
                        submit={jest.fn()}
                    />
                </ThemeProvider>
            </>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
