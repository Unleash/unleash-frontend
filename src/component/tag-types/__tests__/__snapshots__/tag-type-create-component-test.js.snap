// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it supports editMode 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="makeStyles-headerContainer-1"
  >
    <div
      className="makeStyles-headerTitleContainer-5"
    >
      <div
        className=""
      >
        <h2
          className="MuiTypography-root makeStyles-headerTitle-6 MuiTypography-h2"
        >
          Update Tag type
        </h2>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-bodyContainer-2"
  >
    <section
      className="contentSpacing tagTypeContainer"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      >
        Tag types allows you to group tags together in the management UI
      </h6>
      <form
        className="addTagTypeForm contentSpacing"
        onSubmit={[Function]}
      >
        <div
          className="formButtons"
        >
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root"
                  >
                    add
                  </span>
                </span>
                Update
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
             
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="cancel"
            >
              <span
                className="MuiButton-label"
              >
                Cancel
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </section>
  </div>
</div>
`;

exports[`renders correctly for creating 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="makeStyles-headerContainer-1"
  >
    <div
      className="makeStyles-headerTitleContainer-5"
    >
      <div
        className=""
      >
        <h2
          className="MuiTypography-root makeStyles-headerTitle-6 MuiTypography-h2"
        >
          Create Tag type
        </h2>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-bodyContainer-2"
  >
    <section
      className="contentSpacing tagTypeContainer"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      >
        Tag types allows you to group tags together in the management UI
      </h6>
      <form
        className="addTagTypeForm contentSpacing"
        onSubmit={[Function]}
      >
        <div
          className="formButtons"
        >
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <span
                    aria-hidden={true}
                    className="material-icons MuiIcon-root"
                  >
                    add
                  </span>
                </span>
                Create
              </span>
            </button>
             
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="cancel"
            >
              <span
                className="MuiButton-label"
              >
                Cancel
              </span>
            </button>
          </div>
        </div>
      </form>
    </section>
  </div>
</div>
`;
